pkg/build/node/const.go

```diff
@@ -24,225 +24,766 @@ const (
+        - name: install-cni
+          image: calico/cni:v3.6.0
+          command: ["/install-cni.sh"]
+          env:
+            # Name of the CNI config file to create.
+            - name: CNI_CONF_NAME
+              value: "10-calico.conflist"
+            # The CNI network config to install on each node.
+            - name: CNI_NETWORK_CONFIG
+              valueFrom:
+                configMapKeyRef:
+                  name: calico-config
+                  key: cni_network_config
+            # Set the hostname based on the k8s node name.
+            - name: KUBERNETES_NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            # CNI MTU Config variable
+            - name: CNI_MTU
+              valueFrom:
+                configMapKeyRef:
+                  name: calico-config
+                  key: veth_mtu
+            # Prevents the container from sleeping forever.
+            - name: SLEEP
+              value: "false"
+          volumeMounts:
+            - mountPath: /host/opt/cni/bin
+              name: cni-bin-dir
+            - mountPath: /host/etc/cni/net.d
+              name: cni-net-dir
+      containers:
+        # Runs node container on each Kubernetes node.  This
+        # container programs network policy and routes on each
+        # host.
+        - name: calico-node
+          image: calico/node:v3.6.0
+          env:
+            # Use Kubernetes API as the backing datastore.
+            - name: DATASTORE_TYPE
+              value: "kubernetes"
+            # Wait for the datastore.
+            - name: WAIT_FOR_DATASTORE
+              value: "true"
+            # Set based on the k8s node name.
+            - name: NODENAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            # Choose the backend to use.
+            - name: CALICO_NETWORKING_BACKEND
+              valueFrom:
+                configMapKeyRef:
+                  name: calico-config
+                  key: calico_backend
+            # Cluster type to identify the deployment type
+            - name: CLUSTER_TYPE
+              value: "k8s,bgp"
+            # Auto-detect the BGP IP address.
+            - name: IP
+              value: "autodetect"
+            # Enable IPIP
+            - name: CALICO_IPV4POOL_IPIP
+              value: "Always"
+            # Set MTU for tunnel device used if ipip is enabled
+            - name: FELIX_IPINIPMTU
+              valueFrom:
+                configMapKeyRef:
+                  name: calico-config
+                  key: veth_mtu
+            # The default IPv4 pool to create on startup if none exists. Pod IPs will be
+            # chosen from this range. Changing this value after installation will have
+            # no effect. This should fall within --cluster-cidr.
+            - name: CALICO_IPV4POOL_CIDR
+              value: "192.168.0.0/16"
```

Comment:
hmm, looks like we aren't setting one right now. 🤔

Reply:
yeah, that's something I'm trying to understand. 
I've created a multinode cluster with the current Calico configuration and the communication between pods and nodes are working, will try to run the e2e tests to see what happens.

Reply:
From Calico documentation, I think that these variables are a great addon
> | CALICO_IPV4POOL_CIDR | The IPv4 Pool to create if none exists at start up. It is invalid to define this variable and NO_DEFAULT_POOLS. | IPv4 CIDR | 192.168.0.0/16 | |
| CALICO_IPV6POOL_CIDR | The IPv6 Pool to create if none exists at start up. It is invalid to define this variable and NO_DEFAULT_POOLS. | IPv6 CIDR | fd80:24e2:f998:72d6::/64 | |

I still need to explore more Calico or maybe we should reach out to Calico's community to ask for advice

Reply:
Ok, I´ve found how to overcome this, we can leverage k8s and the host-local ipam plugin to avoid to create pools in Calico
By the way, I just hit this https://github.com/projectcalico/calico/issues/2458

Reply:
I had node bringup failures when previously testing this with mulit-node FYI.

Looked into clusterCIDR etc. quite a bit more and options for that. not overly thrilled with any of them jus yet.

> Ok, I´ve found how to overcome this, we can leverage k8s and the host-local ipam plugin to avoid to create pools in Calico

can you elaborate some more on this?

Reply:
Setting CALICO_IPV4POOL_CIDR  to one CIDR implies to set `podSubnet` in `kubeadm` to the same value, creating a dependency between ` kind` and `calico`. Basically, I'm trying to decouple the IPAM configuration from Calico leveraging k8s IPAM and the node podCIDR field annotation.
This way we can have a static Calico configuration and, in the future, being able to add new CNIs without requiring any changes to ` kind`, just only providing an installation manifest.
Another interesting debate is that we should set the ` podSubnet` field, [are we happy predefining its value in ` kind` ? seems that is not clear this point in kubeadm](https://github.com/kubernetes/kubeadm/issues/1011)